src/main/java/com/iwaconsolti/league/demo/controller/MatchController.java

```diff
@@ -0,0 +1,111 @@
+package com.iwaconsolti.league.demo.controller;
+
+import com.iwaconsolti.league.demo.dto.request.MatchCreateRequest;
+import com.iwaconsolti.league.demo.dto.response.MatchResponse;
+import com.iwaconsolti.league.demo.dto.response.TeamSummaryResponse;
+import com.iwaconsolti.league.demo.model.League;
+import com.iwaconsolti.league.demo.model.Match;
+import com.iwaconsolti.league.demo.model.Team;
+import com.iwaconsolti.league.demo.service.LeagueService;
+import com.iwaconsolti.league.demo.service.MatchService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/leagues/{leagueId}/matches")
+@Slf4j
+@RequiredArgsConstructor
+public class MatchController {
+    private final MatchService matchService;
+    private final LeagueService leagueService;
+
+    @PostMapping
+    public ResponseEntity<MatchResponse> createMatch(
+            @PathVariable long leagueId,
+            @RequestBody MatchCreateRequest request) {
+        try {
+            log.info("Creating match in league {}", leagueId);
```

Comment:
Log should be done after the action is completed, and also it should indicate at leas the teams